rust/datafusion/src/execution/aggregate.rs

```diff
@@ -246,42 +246,42 @@ impl AggregateFunction for SumFunction {
     }
 
     fn accumulate_scalar(&mut self, value: &Option<ScalarValue>) {
-        if self.value.is_none() {
-            self.value = value.clone();
-        } else if value.is_some() {
-            self.value = match (&self.value, value) {
-                (Some(ScalarValue::UInt8(a)), Some(ScalarValue::UInt8(b))) => {
-                    Some(ScalarValue::UInt8(*a + b))
-                }
-                (Some(ScalarValue::UInt16(a)), Some(ScalarValue::UInt16(b))) => {
-                    Some(ScalarValue::UInt16(*a + b))
-                }
-                (Some(ScalarValue::UInt32(a)), Some(ScalarValue::UInt32(b))) => {
-                    Some(ScalarValue::UInt32(*a + b))
-                }
-                (Some(ScalarValue::UInt64(a)), Some(ScalarValue::UInt64(b))) => {
-                    Some(ScalarValue::UInt64(*a + b))
-                }
-                (Some(ScalarValue::Int8(a)), Some(ScalarValue::Int8(b))) => {
-                    Some(ScalarValue::Int8(*a + b))
-                }
-                (Some(ScalarValue::Int16(a)), Some(ScalarValue::Int16(b))) => {
-                    Some(ScalarValue::Int16(*a + b))
-                }
-                (Some(ScalarValue::Int32(a)), Some(ScalarValue::Int32(b))) => {
-                    Some(ScalarValue::Int32(*a + b))
-                }
-                (Some(ScalarValue::Int64(a)), Some(ScalarValue::Int64(b))) => {
-                    Some(ScalarValue::Int64(*a + b))
-                }
-                (Some(ScalarValue::Float32(a)), Some(ScalarValue::Float32(b))) => {
-                    Some(ScalarValue::Float32(a + *b))
-                }
-                (Some(ScalarValue::Float64(a)), Some(ScalarValue::Float64(b))) => {
-                    Some(ScalarValue::Float64(a + *b))
-                }
-                _ => panic!("unsupported data type for SUM"),
-            }
+        use ScalarValue::*;
+        self.value = match (&self.value, value) {
+            (None, _) => value.clone(), 
+            (Some(__), None) => unimplemented!(),
```

Comment:
The `unimplemented!()` here, especially without context, doesn't look good. I can see that the current code leaves that code path, so I'm curious as to what would happend if `value` is `None`. Was it panicking?